src/sparseml/yolov8/utils/export_samples.py

```diff
@@ -0,0 +1,174 @@
+    https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/v8/detect/val.py
+    """
+    batch["img"] = batch["img"].to(device, non_blocking=True)
+    batch["img"] = (batch["img"].half() if half else batch["img"].float()) / 255
+    for k in ["batch_idx", "cls", "bboxes"]:
+        batch[k] = batch[k].to(device)
+    return batch
+
+
+@torch.no_grad()
+def export_sample_inputs_outputs(
+    data_loader: torch.utils.data.DataLoader,
+    model: torch.nn.Module,
+    save_dir: str,
+    device: device_class,
+    number_export_samples: int,
+    onnx_path: str,
+):
+    """
+    Export sample model input and output for testing with the DeepSparse Engine
+    :param data_loader: path to data loader to take samples from
+    :param model: model to be exported. Used to generate torch outputs
+    :param save_dir: directory to save samples to
+    :param device: device to run the inference (output generation) on
+    :param number_export_samples: number of samples to export
+    :param onnx_path: path to onnx model. Used to generate ORT outputs
+    """
+
+    LOGGER.info(
+        f"Exporting {number_export_samples} sample model inputs and outputs for "
+        "testing with the DeepSparse Engine"
+    )
+
+    exported_samples = 0
+
+    # Sample export directories
+    sample_in_dir = os.path.join(save_dir, "sample_inputs")
+    sample_out_dir_torch = os.path.join(save_dir, "sample_outputs_torch")
+    sample_out_dir_ort = os.path.join(save_dir, "sample_outputs_onnxruntime")
+
+    os.makedirs(sample_in_dir, exist_ok=True)
+    os.makedirs(sample_out_dir_torch, exist_ok=True)
+    os.makedirs(sample_out_dir_ort, exist_ok=True)
+
+    save_inputs_as_uint8 = _graph_has_uint8_inputs(onnx_path) if onnx_path else False
+
+    # Prepare model for inference
+    model = model.to(device)
+    model.eval()
+
+    # Prepare onnxruntime engine for inference
+    ort_session = onnxruntime.InferenceSession(onnx_path)
+
+    LOGGER.info(f"Exporting sample inputs to directory {sample_in_dir}")
+    LOGGER.info(f"Exporting sample torch outputs to directory {sample_out_dir_torch}")
+    LOGGER.info(
+        f"Exporting sample onnxruntime outputs to directory {sample_out_dir_ort}"
+    )
+
+    for batch in data_loader:
+        file_idx = f"{exported_samples}".zfill(4)
+        preprocessed_batch = preprocess(batch=batch, device=device)
+        image = preprocessed_batch["img"]
+
+        # Save inputs as numpy array
+        _export_inputs(image, sample_in_dir, file_idx, save_inputs_as_uint8)
+        # Save torch outputs as numpy array
+        _export_torch_outputs(image, model, sample_out_dir_torch, file_idx)
+        # Save onnxruntime outputs as numpy array
+        _export_ort_outputs(
+            image.cpu().numpy(), ort_session, sample_out_dir_ort, file_idx
+        )
+
+        exported_samples += 1
+
+        if exported_samples >= number_export_samples:
+            break
+
+    if exported_samples < number_export_samples:
```

Comment:
nit: could put this in a `for .. else` clause? will avoid `exported_samples < number_export_samples` check after the loop